public/friendlycode/js/fc/load-project-files.js

```diff
@@ -0,0 +1,215 @@
+          return;
+        }
+      }
+
+      var length = files.length;
+      var completed = 0;
+      var filePathToOpen;
+
+      function persistFile(path, data, next) {
+        var request = $.ajax({
+          contentType: "application/json",
+          headers: {
+            "X-Csrf-Token": options.csrfToken
+          },
+          type: "PUT",
+          url: options.persistenceURL,
+          data: JSON.stringify({
+            path: path,
+            buffer: data
+          })
+        });
+        request.done(function() {
+          if(request.readyState !== 4) {
+            return;
+          }
+
+          if(request.status !== 201 && request.status !== 200) {
+            // TODO: handle error case here
+            console.error("Server did not persist file");
+            return callback(new Error("Could not persist file"));
+          }
+
+          console.log("Successfully persisted ", path);
+          next();
+        });
+        request.fail(function(jqXHR, status, err) {
+          console.error("Failed to send request to persist the file to the server with: ", err);
+          callback(err);
+        });
+      }
+
+      function checkProjectLoaded() {
+        if(completed === length) {
+          callback(null, {
+            root: root,
+            open: filePathToOpen
+          });
+        }
+      }
+
+      function writeFile(path, data) {
+        var parent = Path.dirname(path);
+
+        function write() {
+          fs.writeFile(path, data, function(err) {
+            if(err) {
+              // TODO: handle error case here
+              console.error("Failed to write: ", path);
+              callback(err);
+              return;
+            }
+
+            if(!options.isNew) {
+              completed++;
+              checkProjectLoaded();
+              return;
+            }
+
+            persistFile(path, data, function() {
+              completed++;
+              checkProjectLoaded();
+            });
+          });
+        }
+
+        shell.mkdirp(parent, function(err) {
+          if(err && err.code !== "EEXIST") {
+            console.error("Failed to create project directory: ", parent);
+            callback(err);
+            return;
+          }
+
+          write();
+        });
+      }
+
+      if(!length) {
+        // TODO: What should we do here? :P
+        callback(new Error("No files to load"));
+        return;
+      }
+
+      files.forEach(function(file) {
+        // TODO: Make this configurable
+        if(!filePathToOpen || Path.extname(filePathToOpen) !== ".html") {
+          filePathToOpen = relative(file.path);
+        }
+
+        writeFile(file.path, new FilerBuffer(file.buffer));
+      });
+    }
+
+    request.onreadystatechange = updateFs;
+    request.responseType = "json";
+    request.open("GET", projectFilesUrl, true);
+    request.send();
+  }
+
+  function generateDefaultProject(title) {
+    return {
+      title: title || "New Project",
```

Comment:
It might be wise to move constants to another module, `Constants.js`, so we can easily modify them without hunting.